### DISCUSSION TITLE
lsp-zero?

### DISCUSSION THREAD

**Discussion by boozedog** [nix-community/nixvim#523] [Q&A] [answered]
I'm trying to match ThePrimeagen's setup from his "Neovim RC" video. He's got the lsp-zero plugin installed, is that possible to install with nixvim? I don't see it in the plugin options.

https://github.com/VonHeikemen/lsp-zero.nvim

**Comment by GaetanLepage** [comment, marked as answer]
Hello !
Currently we don't support lspzero. I don't think it is super useful because nixvim already does some of the work for you when you enable `plugins.nvim-cmp` and `plugins.lsp`. For instance, the `capabilities` stuff regarding the `nvim_lsp` source is already managed automatically.
Also, enabling a specific server is as simple as doing `plugins.lsp.servers.anyServer.enable = true`.
This will also install it for you thanks to nixpkgs (so the `mason` part is useless).

You have two options:
1) (recommended) Configure `nvim-lsp` and `nvim-cmp` "manually" through `plugins.nvim-cmp` and `plugins.lsp`. This is not that hard.
You can have a look at my config as an example: [`completion.nix`](https://github.com/GaetanLepage/dotfiles/blob/be80632010fab858b46fb7b461ee0b3cd6ec4b97/home/modules/tui/neovim/completion.nix) [`lsp.nix`](https://github.com/GaetanLepage/dotfiles/blob/be80632010fab858b46fb7b461ee0b3cd6ec4b97/home/modules/tui/neovim/plugins/lsp.nix).
2) You can also use `lsp-zero` like all of the plugins that we do not "support" (i.e. `plugins.lsp-zero` does not exist) by simply adding it to `extraPlugins`:
```nix
programs.nixvim = {
  extraPlugins = [pkgs.vimPlugins.lsp-zero-nvim];

  extraConfigLua = ''
    local lsp = require('lsp-zero').preset({})

    lsp.on_attach(function(client, bufnr)
      -- see :help lsp-zero-keybindings
      -- to learn the available actions
      lsp.default_keymaps({buffer = bufnr})
    end)

    -- (Optional) Configure lua language server for neovim
    require('lspconfig').lua_ls.setup(lsp.nvim_lua_ls())

    lsp.setup()
  '';
};
```

**Reply by GaetanLepage** [reply]
Were you able to create a working config ?
